workflow/controller/operator.go

```diff
@@ -1117,6 +1124,12 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 		node, err := woc.wf.Status.Nodes.Get(nodeID)
 		if err == nil {
 			if newState := woc.assessNodeStatus(pod, node); newState != nil {
+				// Check whether its taskresult is in an incompleted state.
+				if newState.Succeeded() && woc.wf.Status.IsTaskResultIncomplete(node.ID) {
+					woc.log.WithFields(log.Fields{"nodeID": newState.ID}).Debug("Taskresult of the node not yet completed")
+					podReconciliationCompleted = true
```

Comment:
Unfortunately, you will need to flip all of the true/false here and where it's being checked now - this should represent `podReconciliationCompleted = false`, right?

Reply:
Actually, here I think it would be fine to just name it back to `taskResultIncomplete` - maybe I can try do it inline everywhere and you can see what I mean. (Sorry for the trouble!)

Reply:
Okay, I just added inline suggestions everywhere - what do you think? Does it make sense? Hopefully, I don't have any errors.

Reply:
> Okay, I just added inline suggestions everywhere - what do you think? Does it make sense? Hopefully, I don't have any errors.

Yes, you are right! I have accept the suggestions.